action/src/plugins/imageDiff/plugin.ts

```diff
@@ -0,0 +1,80 @@
+import * as core from '@actions/core';
+import { randomUUID } from 'crypto';
+import fs from 'fs';
+import download from 'image-downloader';
+import { basename } from 'path';
+import pixelmatch from 'pixelmatch';
+import { PNG } from 'pngjs';
+
+import { ImageDiffPlugin } from '../../schemas/stoatConfigSchema';
+import { GithubActionRun, UploadImageDiffRequest } from '../../types';
+import { submitPartialConfig } from '../helpers';
+import { createSignedUrl, uploadPath } from '../staticHosting/helpers';
+
+const runImageDiffPlugin = async (
+  taskId: string,
+  taskConfig: ImageDiffPlugin,
+  { ghToken, ghRepository: { repo, owner }, ghSha }: GithubActionRun,
+  stoatConfigFileId: number
+) => {
+  core.info(`[${taskId}] Running static hosting plugin (stoat config ${stoatConfigFileId})`);
+  const currentDirectory = process.cwd();
+  core.info(`[${taskId}] Current directory: ${currentDirectory}`);
+
+  // get baseline image
+  if (!taskConfig.baseline) {
+    throw new Error(`[${taskId}] Missing baseline image URL in ${taskId} task config`);
+  }
+  const uuid = randomUUID();
+  const baselinePath = `${currentDirectory}/${uuid}-baseline.png`;
+  core.info(`[${taskId}] Downloading baseline image to ${baselinePath}...`);
+  const baselineImageFile = await download.image({
```

Comment:
Separate from image diffing specifically, I think introducing a dependency for pulling data from an external source is a pretty significant decision, since there are likely permissioning-related issues for all private repo use cases. Added a note for us to discuss this.

Reply:
Good point.